corehq/apps/products/views.py

```diff
@@ -355,6 +355,7 @@ def product_importer_job_poll(request, domain, download_id,
     return render(request, template, context)
 
 
+@login_and_domain_required
```

Comment:
The product list view requires domain admin: https://github.com/dimagi/commcare-hq/blob/2c48946e7eff3a047fc6a2fd39cbb49fa2131a4d/corehq/apps/commtrack/views.py#L60-L62

But this does take care of the main risk.